common/nodejs-utils/src/config.rs

```diff
@@ -0,0 +1,338 @@
+use crate::error_handling::ErrorType::UserFacing;
+use crate::error_handling::{
+    error_message, ErrorType, SharedErrorMessageBuilder, SuggestRetryBuild, SuggestSubmitIssue,
+};
+use bullet_stream::style;
+use indoc::formatdoc;
+use std::path::Path;
+use std::str::FromStr;
+use toml_edit::TableLike;
+
+const CONFIG_NAMESPACE: [&str; 5] = ["com", "heroku", "buildpacks", "nodejs", "actions"];
+const PRUNE_DEV_DEPENDENCY_CONFIG: &str = "prune_dev_dependencies";
+
+pub fn read_prune_dev_dependencies_from_project_toml(
+    project_toml_path: &Path,
+) -> Result<Option<bool>, ConfigError> {
```

Comment:
Suggestion (not blocking): I see an opportunity to reduce/remove the need to create tempfiles for tests, if the interface is split in two:

- `fn toml_edit_from_path(path: &Path) -> Result<Option<toml_edit::DocumentMut>, ConfigError> {`
- `fn dev_dependencies_config(doc: &toml_edit::DocumentMut) -> Result<Option<bool>, ConfigError> {`

You would still need tempfiles to exercise `toml_edit_from_path`, but the testing of the `dev_dependencies_config` could take a direct input that can be constructed without file creation.

The type signature return of the first (`toml_edit_from_path`) forces you to handle the "file does not exist" externally.